### ISSUE TITLE
Static File cache not invalidating sites with path

### ISSUE THREAD

**Issue by blackshadev** [statamic/cms#7403]
### Bug description

We use statamic with multiple sites on the same domain with different paths. The file caching itself it works great, even with changing a page since it uses the absoluteURL of the entry it works.

But when using invalidation rules it will use the wrong 'domain'  for `getPathAndDomain`.

During debugging I notices the following two issues: 

First, the getBaseURL `src/vendor/statamic/cms/src/StaticCaching/Cachers/AbstractCacher.php` doesn't work when called from the control panel. It returns the Site::Current. but it should be Site::Selected. Otherwise you will clear the wrong command.

Second, It seems like there are two types of code paths which access `getPathAndDomain`:
- One with an absolute URL 
- One with a relative URL

For the absolute url it will parse the url and return the domain as `{scheme}://{host}`
For the relative one it will use the baseUrl as domain, which isn't if your baseUrl includes a path. For instance `http://example.com/nl`

This means that the code paths behave different and clear / cache differently.

### How to reproduce

Have a statamic environment with multiple sites on the same domain

```
return [
    'sites' => [
        'enrise_com' => [
            'name' => 'Enrise.com',
            'slug' => 'enrise_com',
            'locale' => 'en_US',
            'url' => 'http://enrise.local'
            'origin' => null,
        ],
        'enrise_com_nl' => [
            'name' => 'Enrise.com',
            'slug' => 'enrise_com_NL',
            'locale' => 'nl_NL',
            'url' => 'http://enrise.local/nl'
            'origin' => 'enrise_com',
        ],
    ] ,
];
```

enable static file cache in .env
```
STATAMIC_STATIC_CACHING_STRATEGY=full
```

Set up invalidation rules and strategies in `config/statamic/static_caching.php`
```
    'strategies' => [

        'half' => [
            'driver' => 'application',
            'expiry' => null,
        ],

        'full' => [
            'driver' => 'file',
            'path' => public_path('static/enrise'),
            'lock_hold_length' => 0,
        ],

    ],
    'invalidation' => [

        'class' => null,

        'rules' => [
            'navigation' => [
                'main_navigation' => [
                    'urls' => ['/*'],
                ],
                'footer_navigation' => [
                    'urls' => ['/*'],
                ]
            ]
        ],

    ],
```

Load some content into it with the main navigation on top

Navigate to the NL site. See it cached.

Change the navigation

See the cache isn't cleared.

Change a NL entry (homepage)

See the cache cleared (because it uses the entry absolute path)

### Environment

```yaml
Environment
Application Name: Statamic
Laravel Version: 9.33.0
PHP Version: 8.1.13
Composer Version: 2.4.1
Environment: local
Debug Mode: ENABLED
URL: dayes.local
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: pgsql
Logs: stack / single, stderr
Mail: smtp
Queue: sync
Scout: database
Session: database

Statamic
Addons: 6
Antlers: regex
Stache Watcher: Enabled
Static Caching: full
Version: 3.3.65 PRO

Statamic Addons
enrise/portal: dev-main
enrise/products: dev-main
mitydigital/sitemapamic: 2.3.3
rias/statamic-button-box: 2.0.3
rias/statamic-redirect: 2.5.0
swiftmade/statamic-clear-assets: dev-modified-date-check
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

None

**Comment by arthurperton** [comment]
Hi @blackshadev, while recreating your issue I actually got a toast saying "Undefined array key '...'" when saving the nav. Looking at the logs it is exactly the same error as in #7411. 

Applying the PR for that one (#7509) fixed your issue for me. Are you able to give that a try?

**Comment by blackshadev** [comment]
Hi @arthurperton I tried #7509, but it didn't solve the problem. I also didn't have this "Undefined array key" error. 

My reasoning still seems valid, the getPathAndDomain function is called in 3 ways: 
- With an absolute path (when invalidating an entry with an URL), this works fine.
- With a relative path from a page fetch where the Site::current() is used correctly as the page is of the current site. 
- And with a relative path from the Control Panel when invalidating anything else. Here Site::current() will return the "wrong" site, as you are currently in the ControlPanel and not the site you are editing (Site::selected()). These two can have wildly different domains and thus different static-caches.

I tried changing things in the vendor folder, but my Statamic knowledge is to limited to know if I will break something, more so since getPathAndDomain  seems to be called wrong allot of different contexts.
